src/components/GameOver.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useState, useEffect } from "react";
+import Modal from "react-modal";
+import styled from "styled-components";
+import { useSelector, useDispatch } from "react-redux";
+import { State, store } from "../../public/store";
+
+interface ModalProps {
+  isOpen: boolean;
+}
+const GameOverModal = styled(Modal)<ModalProps>``;
+const Heading = styled.h2``;
+const Winner = styled.h3``;
+const StyledButton = styled.button``;
+
+interface GameOverProps {
+  ws: WebSocket;
+}
+
+const GameOver: React.FC<GameOverProps> = ({ ws }) => {
+  const [modalIsOpen, setModalIsOpen] = useState(false);
+  const winner = useSelector((state: State) => state.winner);
+  const dispatch = useDispatch();
+  const setModalIsOpenToTrue = () => {
+    setModalIsOpen(true);
+  };
+  const setModalIsOpenToFalse = () => {
+    setModalIsOpen(false);
+  };
+
+  GameOverModal.setAppElement("body");
```

Comment:
I'm not completely sure how this works/what it's for, but I have a general mistrust of calling random DOM-feeling things inside the render function so I did a touch of googling.

In their example at https://reactcommunity.org/react-modal/examples/set_app_element/ they call it at the top level (not inside any function). They also refer to the same element that they pass into `ReactDOM.render()` at the bottom.

I can't find proper docs for this though, so I'm probably chatting shit.